src/AppInstallerCLICore/Commands/DscUserSettingsResource.cpp

```diff
@@ -0,0 +1,258 @@
+                    if (target.isMember(overlayKey))
+                    {
+                        Json::Value& targetValue = target[overlayKey];
+                        if (targetValue.isObject() && overlayValue.isObject())
+                        {
+                            // Recursively merge the objects.
+                            MergeUserSettings(targetValue, overlayValue);
+                        }
+                        else
+                        {
+                            // Replace the value in the target.
+                            // Note: Arrays are not merged, they are replaced.
+                            target[overlayKey] = overlayValue;
+                        }
+                    }
+                    else
+                    {
+                        // Add the overlay key to the target.
+                        target[overlayKey] = overlayValue;
+                    }
+                }
+            }
+        };
+    }
+
+    DscUserSettingsResource::DscUserSettingsResource(std::string_view parent) :
+        DscCommandBase(parent, "user-settings", DscResourceKind::Resource,
+            DscFunctions::Get | DscFunctions::Set | DscFunctions::Test | DscFunctions::Export | DscFunctions::Schema,
+            DscFunctionModifiers::ImplementsPretest | DscFunctionModifiers::HandlesExist | DscFunctionModifiers::ReturnsStateAndDiff)
+    {
+    }
+
+    Resource::LocString DscUserSettingsResource::ShortDescription() const
+    {
+        return Resource::String::DscUserSettingsShortDescription;
+    }
+
+    Resource::LocString DscUserSettingsResource::LongDescription() const
+    {
+        return Resource::String::DscUserSettingsLongDescription;
+    }
+
+    std::string DscUserSettingsResource::ResourceType() const
+    {
+        return "WinGetUserSettings";
+    }
+
+    void DscUserSettingsResource::ResourceFunctionGet(Execution::Context& context) const
+    {
+        ResourceFunctionExport(context);
+    }
+
+    void DscUserSettingsResource::ResourceFunctionSet(Execution::Context& context) const
+    {
+        if (auto json = GetJsonFromInput(context))
+        {
+            UserSettingsFunctionData data{ json };
+
+            if (data.Load())
+            {
+                data.Set();
+                if (!data.Test() && !data.Write())
+                {
+                    AICLI_LOG(Config, Error, << "Failed to write output to user settings file.");
+                    return;
+                }
+
+                WriteJsonOutputLine(context, data.Output.ToJson());
+                WriteJsonOutputLine(context, data.DiffJson());
+            }
+        }
+    }
+
+    void DscUserSettingsResource::ResourceFunctionTest(Execution::Context& context) const
+    {
+        if (auto json = GetJsonFromInput(context))
+        {
+            UserSettingsFunctionData data{ json };
+
+            if (data.Load())
+            {
+                data.Set();

```

Comment:
Again, this is **WILD** that `Test` (a read only action) calls `Set` (ostensibly a write action).